### PR TITLE
Add host time platform; remove host support from sntp.

### PR DESCRIPTION
**Pull request by clydebarrow**
# What does this implement/fix?

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#3909

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx
- [x] host

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/sntp/time.py [hunk 1]
```diff
@@ -2,24 +2,31 @@
...
+CONFIG_SCHEMA = cv.All(
+    time_.TIME_SCHEMA.extend(
+        {
+            cv.GenerateID(): cv.declare_id(SNTPComponent),
+            cv.Optional(CONF_SERVERS, default=DEFAULT_SERVERS): cv.All(
+                cv.ensure_list(cv.Any(cv.domain, cv.hostname)), cv.Length(min=1, max=3)
+            ),
+        }
+    ).extend(cv.COMPONENT_SCHEMA),
+    cv.only_on([PLATFORM_ESP32, PLATFORM_ESP8266, PLATFORM_RP2040]),
```
Comment (jesserockz) [inline esphome/components/sntp/time.py]: Doesnt this work fine on the two libretiny platforms?

Comment (clydebarrow) [reply esphome/components/sntp/time.py]: Probably, but it's not tested for or specifically mentioned in the CPP code.

Comment (clydebarrow) [reply esphome/components/sntp/time.py]: Tested, and working on bk72xx. I don't have rtl87xx to test on but it's safe to assume it will work I think.

Comment (jesserockz) [reply esphome/components/sntp/time.py]: > Probably, but it's not tested for or specifically mentioned in the CPP code.

It was mentioned before #6769 =)

**Review by jesserockz — APPROVED** [review]

**Comment by nagyrobi** [comment]
Works, thanks.

**Pull request merged**